src/vsphere_cpi/lib/cloud/vsphere/cloud.rb

```diff
@@ -15,16 +17,18 @@ def message
       end
     end
 
-    attr_accessor :client, :logger # exposed for testing
+    attr_accessor :client
     attr_reader :config, :datacenter, :heartbeat_thread
 
     def enable_telemetry
       http_client = VSphereCloud::CpiHttpClient.new
       other_client = VCenterClient.new(
         vcenter_api_uri: @config.vcenter_api_uri,
-        http_client: http_client,
-        logger: Logger.new('/dev/null'),
+        http_client: http_client
       )
+      def other_client.logger
```

Comment:
Seeing def inside def doesnt look nice. Could we instead do other_client.define_sigleton_method(:logger) ?